pepper/test/apps/trade_execution_test.cpp

```diff
@@ -82,62 +81,93 @@ TEST(TradeExecutionTest, ValidSingleOrderTrade) {
+    orders[0].account = 0;
+    orders[0].sellToken = 1;
+    orders[0].buyToken = 2;
+    orders[0].sellAmount = 100;
+    orders[0].buyAmount = 99;
 
+    // second order
+    orders[1].account = 1;
+    orders[1].sellToken = 2;
+    orders[1].buyToken = 1;
+    orders[1].sellAmount = 100;
+    orders[1].buyAmount = 99;
+
+    // Volumes only roughly match prices,
+    // we are selling one more unit than we buy
+    // and providing less suprlus
+    volumes[0].sellVolume = 100;
+    volumes[0].buyVolume = 99;
+    volumes[0].surplus = 0;
+    volumes[1].sellVolume = 100;
+    volumes[1].buyVolume = 100;
+    volumes[1].surplus = 0;
+
+    //Make sure there is balance
+    balances[1] = volumes[0].sellVolume;
+    balances[TOKENS + 2] = volumes[1].sellVolume;
+
+    //Provide private input
     privateInput = serializePrivateInput(balances, orders, volumes, prices);
-    field254 state = hashPedersen(privateInput.balances, 0, ACCOUNTS*TOKENS*BITS_PER_DECIMAL, BITS_PER_DECIMAL);
+
     ShaResult hashBatchInfo = hashSHA(privateInput.pricesAndVolumes, 0, (TOKENS*BITS_PER_DECIMAL)+(ORDERS*BITS_PER_DECIMAL), 256);
     struct In input {
-        state + 1,
+        hashPedersen(privateInput.balances, 0, ACCOUNTS*TOKENS*BITS_PER_DECIMAL, BITS_PER_DECIMAL), // state
         0, //surplus
         {hashBatchInfo.left, hashBatchInfo.right}, //hashBatchInfo
-        hashPedersen(privateInput.orders, 0, ORDERS*BITS_PER_ORDER, BITS_PER_ORDER),
-        1 //one
+        hashPedersen(privateInput.orders, 0, ORDERS*BITS_PER_ORDER, BITS_PER_ORDER)
     };
     struct Out output;
+    compute(&input, &output);
 
-    DISABLE_STACKTRACE = true;
-    ASSERT_DEATH(compute(&input, &output), "");
-    DISABLE_STACKTRACE = false;
+    // Execute trade and check that output matches
+    balances[1] = 0;
+    balances[2] = 99;
+    balances[TOKENS + 1] = 100;
+    balances[TOKENS + 2] = 0;
+    field254* updatedBalances = serializePrivateInput(balances, orders, volumes, prices).balances;
+    field254 updatedHash = hashPedersen(updatedBalances, 0, ACCOUNTS*TOKENS*BITS_PER_DECIMAL, BITS_PER_DECIMAL);
+    ASSERT_EQ(output.state, updatedHash);
 }
 
-TEST(TradeExecutionTest, InvalidOrderHash) {
-    struct Balance balances[ACCOUNTS] = {0};
+#pragma mark - Invalid instances
+
+TEST(TradeExecutionTest, InvalidBalanceHash) {
+    field254 balances[TOKENS * ACCOUNTS] = {0};
     struct Order orders[ORDERS] = {0};
     struct Volume volumes[ORDERS] = {0};
     field254 prices[TOKENS] = {0};
 
     privateInput = serializePrivateInput(balances, orders, volumes, prices);
     field254 state = hashPedersen(privateInput.balances, 0, ACCOUNTS*TOKENS*BITS_PER_DECIMAL, BITS_PER_DECIMAL);
-    field254 orderHash = hashPedersen(privateInput.orders, 0, ORDERS*BITS_PER_ORDER, BITS_PER_ORDER);
     ShaResult hashBatchInfo = hashSHA(privateInput.pricesAndVolumes, 0, (TOKENS*BITS_PER_DECIMAL)+(ORDERS*BITS_PER_DECIMAL), 256);
     struct In input {
-        state,
+        state + 1,
```

Comment:
What is this state + 1? Is this the incrementing integer of state transitions?

Reply:
It's incremented to make it invalid. state + 1 != state

This is a new test case, rather than a change in an existing one (opposed to what the diff view suggests). Because the tests are similar, it looks like git is trying to match this code with the previous test cases.